### ISSUE TITLE
dartfmt does not correct indent one-per-line argument list with @Inject()

### ISSUE THREAD

**Issue by haejeong87** [dart-lang/dart_style#444]
Given the following correctly formatted Dart code
```dart
class MyComponent {
  Object firstArgument;
  Object superDuperLongNamedArgument;
  Object secondArgument;

  MyComponent(
      @Inject(const MyFirstArgument()) this.firstArgument,
      @Inject(const MySuperDuperLongNamedArgument())
          this.superDuperLongNamedArgument, // LOOK AT ME
      @Inject(const MySecondArgument()) this.secondArgument);
}
```

dartfmt deindents
```dart
          this.superDuperLongNamedArgument, // LOOK AT ME
```

Resulting
```dart
class MyComponent {
  Object firstArgument;
  Object superDuperLongNamedArgument;
  Object secondArgument;

  MyComponent(
      @Inject(const MyFirstArgument()) this.firstArgument,
      @Inject(const MySuperDuperLongNamedArgument())
      this.superDuperLongNamedArgument, // LOOK AT ME
      @Inject(const MySecondArgument()) this.secondArgument);
}
```

**Comment by munificent** [comment]
This is working as intended. The parameter isn't a subexpression of the annotation (if anything, the converse is true), so I don't think it helps to indent it here.

**Comment by bernardyen** [comment]
Bob,

I would have to disagree with you on this. Many members of our team has ran into it, and all of us believe the correct formatting is with the indentation. In essence, the "this.." line is a continuation of the @Inject notation, and therefore should take +4.

**Comment by haejeong87** [comment]
Hi Bob,
As Bernard mentioned in the comment, I believe +4 is necessary not because it is a subexpression but because it is a continuation of the previous line.
At least, with the +4 indentation, it is easier to tell how many arguments are there.

**Comment by lrhn** [comment]
Most annotations are put at a line preceding the declaration they annotate, and at the same indentation level. That's usually not a problem because there are empty lines between declarations.
For parameters, and parameters only, it seems the general use is to put annotations on the same line as the parameter. That makes sense the way parameters are usually written - you don't have an extra line above and room for whitespace in-between.
With that as a normal layout, I also think the break here should be considered a line continuation and cause extra indentation. 

(I started out writing this message thinking that the formatter is correct, but I managed to talk myself into now thinking it's wrong).

(I also think the line length limit should be increased or removed).
